### PR TITLE
Module Fixings&FollowUps

### PR DESCRIPTION
**Pull request by nupplaphil**
- Make `$_REQUEST` processing independent of sub-calls
- Make `HTTPInputData`dynamic (for better testability)
- Make API call permission checks more reliable (no more forced to call the corresponding `parent::`)

Bugfixings:
- [x] #11023 
- [x] https://github.com/friendica/friendica/pull/11042#issuecomment-981086772

Replaces #11038

### CODE REVIEW

File: src/App.php [hunk 1]
```diff
@@ -702,8 +704,12 @@ public function runFrontend(App\Router $router, IManagePersonalConfigValues $pco
 				$module = $router->getModule();
 			}
 
+			// Processes data from GET requests
+			$httpinput = $httpInput->process();
+			$input     = array_merge($httpinput['variables'], $httpinput['files'], $request ?? $_REQUEST);
+
 			// Let the module run it's internal process (init, get, post, ...)
-			$response = $module->run($_POST, $_REQUEST);
+			$response = $module->run($input);
```
Comment (nupplaphil) [inline src/App.php:712]: @annando - Here's the PHP Input stream processing now. It's called **before** any sub-module is called, so it's independent of PUT/DELETE/POST/GET sub-calls and now delivered with every call as `$request` parameter

**Comment by nupplaphil** [comment]
@annando - can you check https://github.com/friendica/friendica/pull/11044/commits/09fda072385c2e8d306d4daa9b616b7d44c27e81 and https://github.com/friendica/friendica/pull/11044/commits/19603e228a56fdf3f54d9c0b82acaa26ecfa972f ?

This solves #11023 , but I had to remove `'type' => 'photo' at both cases

and https://github.com/friendica/friendica/pull/11044/commits/a89dd7059a53b7962c5396f7f507c538c53735d8 replaces the `DI::apiResponse()->exit(...)` calls, but there **is** this method .. see https://github.com/friendica/friendica/blob/2021.12-rc/src/DI.php#L445

**Comment by annando** [comment]
@nupplaphil I'm away until the evening. I try to have a look at it then.

**Comment by annando** [comment]
@nupplaphil does the `exit` now exits again? In the current version it just adds content to the output but doesn't halt the execution.

Concerning handling the items. Here they are created: https://github.com/friendica/friendica/blob/2021.12-rc/mod/photos.php#L372-L391 So you could have a look at how to search for these posts.

**Comment by nupplaphil** [comment]
> Why have you removed the `type` field?
> 
> Edit: I just realized that the field doesn't exists anymore. But you could check for the `post-type` set to `Item::PT_IMAGE` and `origin` set to `true`.

I'll change it, that was the missing hint, thx

**Comment by nupplaphil** [comment]
> @nupplaphil does the `exit` now exits again? In the current version it just adds content to the output but doesn't halt the execution.

Aaahh I see. No the exit doesnt exit :D .. I'll rename the method. Concerning exiting => the exiting is triggered by the Module execution itself and not more by any subclass. In the end, every module execution returns a `ResponseInterface` with all necessary informationa to just print the response. So there's no more native `exit` calls during execution, just an output with different types :)

**Comment by MrPetovan** [comment]
From #11023, throwing an `HTTPException` in an API module results in HTML errors being sent back in the response. I'd like us to find a way to guess the intended response format (HTML, JSON or XML) in a central place so that the Exception handler module can return the expected content type.

File: src/Module/Api/Friendica/Photo/Delete.php [hunk 2]
```diff
@@ -33,13 +33,13 @@
  */
 class Delete extends BaseApi
 {
-	protected function rawContent(array $request = [])
+	protected function post(array $request = [])
```
Comment (MrPetovan) [inline src/Module/Api/Friendica/Photo/Delete.php]: This endpoint still needs to accept DELETE requests.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Comment by annando** [comment]
@nupplaphil when you rename the `exit` then please also change the code at the places where it is called since there are occurrences where the program execution needs to stop.

**Comment by nupplaphil** [comment]
> @nupplaphil when you rename the `exit` then please also change the code at the places where it is called since there are occurrences where the program execution needs to stop.
> 

Already done :)

**Comment by MrPetovan** [comment]
Please merge #11037 first and rebase this larger PR on top of it, please note I had to fiddle with the `BaseModule->run` calls in the tests to skip the redundant `$post` parameter.

**Comment by MrPetovan** [comment]
I went ahead and rebased this PR, pruning the commits I cherry-picked in #11037 and updating the tests with the changes to `BaseModule->run` parameters.

**Review by MrPetovan — APPROVED** [review]

**Comment by nupplaphil** [comment]
> I went ahead and rebased this PR, pruning the commits I cherry-picked in #11037 and updating the tests with the changes to `BaseModule->run` parameters.

Thx! :)

**Comment by nupplaphil** [comment]
> I went ahead and rebased this PR, pruning the commits I cherry-picked in #11037 and updating the tests with the changes to `BaseModule->run` parameters.

Thx! :)

**Comment by MrPetovan** [comment]
As a general rule, I have more available time more frequently than you seem to do, so don't hesitate to open stub PRs with instructions for how to complete it, especially if it's about sweeping changes, and I can pick them up so we don't have to wait for your next availability.

**Pull request merged**